src/components/Recorder/Recorder.tsx

```diff
@@ -45,18 +45,13 @@ export class Recorder extends Component<Props, State> {
         this.screenBlobs = [];
         this.startTime = 0;
     }
+    componentDidMount() {
+        this.startStream();
+    }
 
     componentDidUpdate(prevProps: Props) {
-        const { stream, screenStream, doRecording, discard, doPlayback } = this.props;
-        if (!doPlayback) {
-            if (this.videoRef && stream) {
-                this.videoRef.srcObject = stream;
-            }
-
-            if (screenStream && this.screenRef) {
-                this.screenRef.srcObject = screenStream;
-            }
-        }
+        const { doRecording, discard } = this.props;
+        this.startStream();
```

Comment:
do you really want to `startStream` any time the component is updated?

Reply:
yes, because I don't know what was changed in the streams - can be resource, or specific device...

Reply:
technically I can compare between prevPropv but I don't think it worth the effort of comparing those complicated objects